platform-validation/scripts/analyzeSystemStats.sh

```diff
@@ -0,0 +1,245 @@
+          else
+               echo "$(date): Writing avg $type usage for $binary, Value: $value"
+               echo "$binary-avg-$type=$value" >>"$FILE"
+          fi
+     fi
+
+}
+
+perform_analysis() {
+     IOTEDGE_BINARIES_SIZE=0.0
+     IOTEDGE_BINARIES_MEMORY=0.0
+     IOTEDGE_CONTAINERS_SIZE=0.0
+     IOTEDGE_CONTAINERS_MEMORY=0.0
+
+     for binary in $BINARIES; do
+          if [[ $binary =~ aziot-* ]]; then
+               binary_size="$(grep "$binary"-size <"$1" | sed -r "s/$binary-size=//g")"
+               memory_usage="$(grep "$binary"-avg-memory <"$1" | sed -r "s/$binary-avg-memory=//g")"
+               IOTEDGE_BINARIES_SIZE=$(echo "$IOTEDGE_BINARIES_SIZE" "$binary_size" | awk '{print $1 + $2}')
+               IOTEDGE_BINARIES_MEMORY=$(echo "$IOTEDGE_BINARIES_MEMORY" "$memory_usage" | awk '{print $1 + $2}')
+          fi
+     done
+
+     if [[ -n "$CONTAINERS" ]]; then
+
+          stored_total_container_size="$(grep "total-container-size" <"$1" | sed -r "s/total-container-size=//g")"
+          if [[ -n $stored_total_container_size ]]; then
+               for container in $CONTAINERS; do
+                    if [[ $container =~ edgeHub ]] || [[ $container =~ edgeAgent ]]; then
+                         memory_usage="$(grep "$container"-avg-memory <"$1" | sed -r "s/$container-avg-memory=//g")"
+                         IOTEDGE_CONTAINERS_MEMORY=$(echo "$IOTEDGE_CONTAINERS_MEMORY" "$memory_usage" | awk '{print $1 + $2}')
+                    else
+                         nonruntime_container_size="$(grep "$container"-size <"$1" | sed -r "s/$container-size=//g")"
+                         IOTEDGE_CONTAINERS_SIZE=$(echo "$stored_total_container_size" "$nonruntime_container_size" | awk '{print $1 - $2}')
+                    fi
+               done
+          else
+               echo "Total Container Size should be present, exiting"
+               exit 1
+          fi
+     fi
+
+     echo "iotedge-binaries-size=$IOTEDGE_BINARIES_SIZE"
+     echo "iotedge-binaries-avg-memory=$IOTEDGE_BINARIES_MEMORY"
+     echo "iotedge-container-size=$IOTEDGE_CONTAINERS_SIZE"
+     echo "iotedge-container-memory=$IOTEDGE_CONTAINERS_MEMORY"
+}
+
+process_args "$@"
+if [[ -n $MEMORY_FILE_PATH ]]; then
+     if [[ -f $MEMORY_FILE_PATH ]]; then
+          perform_analysis "$MEMORY_FILE_PATH"
+          exit 0
+     else
+          echo "File $MEMORY_FILE_PATH does not exist"
+          exit 1
+     fi
+fi
+
+echo "Running Usage Test for $SECONDS_TO_RUN seconds"
+FILE="$STORAGE_PATH/usage-$(uname -m).txt"
+echo "Storing Data at $FILE"
+echo "Binaries : $BINARIES"
+echo "Binary Locations : $BINARYLOCATIONS"
+echo "Containers: $CONTAINERS"
+echo "Runtime_Seconds=$SECONDS_TO_RUN"
+echo "OS=$(grep PRETTY_NAME <"/etc/os-release" | sed -r 's/^PRETTY_NAME=//g')"
+echo "ARCH=$(uname -m)"
+echo "Pruning all unused images"
+sudo docker system prune --all --force
+
+end_time=$((SECONDS + SECONDS_TO_RUN))
+while [[ $SECONDS -lt $end_time ]]; do
```

Comment:
Does this run in a tight loop, and is that what we want? I don't see a sleep anywhere...

Reply:
Is there a reason for introducing sleep? I am not continuously logging to a file and only record an increase/change in the stats.

Reply:
>  I was just worried about this stats loop starving other processes (like the edge runtime) that need the CPU.

Not necessarily. I was just worried about this stats loop starving other processes (like the edge runtime) that need the CPU. If running these commands in a loop isn't too CPU intensive, then this should be fine.

Reply:
I thought a bit more about your idea, and I added a configuration to specify a poll interval (in seconds) to the script